py-rse-index.Rmd

```diff
@@ -175,15 +175,10 @@ with all subsequent files created and stored within it:
     ├── sense_and_sensibility.txt
     ├── sherlock_holmes.txt
     └── time_machine.txt
+```
 
-### What you'll learn
-
-In the process of writing and publishing a Python package for Zipf's law verification,
-we cover the tools and practices needed to create work that is open, reproducible, and sustainable.
-You can use this book for self-study
-or as the foundation of a one-semester course for graduate students or senior undergraduates.
-By its end,
-you will be able to:
+In the process of writing and publishing a Python package to verify Zipf's Law,
+we will show you how to:
```

Comment:
The list of things we will show you isn't in the order in which we teach things (e.g. git branching isn't until chapter 6) and some of the things in the list aren't a major focus of what we teach (e.g. working on remote computers).

Here's my attempt at a list that better reflects to contents of the book:
- Use the unix shell to efficiently manage your data and code
- Use Python to write programs that can be executed at the command line
- Use version control software (Git) to keep a revision history of your code
- Use git branches to coordinate changes to your code
- Use an external hosting service (GitHub) to backup/share your code and to manage contributions from others
- Use a build tool (Make) to automate complex data processing workflows
- Enable users to configure your software without modifying it directly
- Efficiently investigate bugs and handle different types of errors in your code
- Work productively in a small team where everyone is welcome
- Effectively review code, including for Python style choices that can make programs more flexible and readable
- Organise your data science projects effectively
- Create Python packages that can be installed in standard ways and be used by others
- Test your software and know which parts have not yet been tested
- Publish your code and research findings in an open and reproducible manner